app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -25,18 +25,30 @@ def create_sub_plans
         job_invocation = JobInvocation.find(input[:job_invocation_id])
         proxy_selector = RemoteExecutionProxySelector.new
 
-        # composer creates just "pattern" for template_invocations because target is evaluated
-        # during actual run (here) so we build template invocations from these patterns
-        job_invocation.targeting.hosts.map do |host|
+        ::Host.where(:id => current_batch).map do |host|
```

Comment:
Right, forgot about it. I fixed it in the [foreman-tasks PR](https://github.com/theforeman/foreman-tasks/pull/229). Since `RunHostsJob` inherits from `Actions::ActionWithSubPlans` we'll get two fixes for the price of one